pkg/cpuallocator/allocator.go

```diff
@@ -157,16 +155,16 @@ func (s *sysfsSingleton) pick(idSlice []sysfs.ID, f IDFilter) []sysfs.ID {
 	return ids[0:idx]
 }
 
-func (s *sysfsSingleton) discoverPriorityCpus() {
-	s.priorityCpus = cpuset.NewCPUSet()
-	if s.sys == nil {
+func (ca *cpuAllocator) discoverPriorityCpus() {
+	ca.priorityCpus = cpuset.NewCPUSet()
+	if ca.sys == nil {
```

Comment:
Is this an error case or is `sys` expected to be empty in this situation? If this is an error, you could check this already in `NewCPUAllocator()`.

Reply:
Thanks for looking this! The original code is written so that sys could be `nil`, in theory. Now that I took a closer look that actually seems to cause a crash. I'll update the PR with revised error handling...

Reply:
#201 addresses the issue